package.json

```diff
@@ -200,8 +200,9 @@
     "@hapi/call": "^8.0.1",
     "@hapi/subtext": "^7.0.3",
     "@kubernetes/client-node": "^0.16.1",
-    "@ogre-tools/injectable": "5.1.2",
-    "@ogre-tools/injectable-react": "5.1.2",
+    "@ogre-tools/injectable": "5.2.0",
+    "@ogre-tools/injectable-react": "5.2.0",
+    "@ogre-tools/fp": "5.2.0",
```

Comment:
I would like to raise notice against using this. I find that even with types `pipeline` is very hard to read. I would like to motion that chaining is better ergonomically.

Reply:
Opinion noted! :) 

Given a bit of time and familiarity the pattern of `pipeline` will once again present itself as **very easy** to read, instead of anything else. This will become more apparent in more complex algorithms.

As with most design patterns, truly realized ergonomy comes from repetition ;)

Noteworthy: `pipeline` has multiple advantages over the proposed **Collection chaining** -pattern:

1. Better support for functional composition, and other functional patterns, such as `flow` or ternary functions.
2. Better refactorability, especially for nested or recursed scenarios.
3. Power to control flow and aspect: eg. if required, `pipeline` can abstract promises and even generators.
4. Full control of vocabulary (and not just `map`, `reduce`, `filter`, etc.) leads to more expressive code.
5. Etc., probably.

We used **Collection chaining** for years before growing out of it. As with everything, we do not use something like `pipeline` to flex, but instead for the necessity of the  _Greater good_ ;)

Reply:
I think the pipeline syntax is good, but I'm a bit wary of adding this particular dependency, which appears to be quite a thin wrapper over lodash. This dependency has really small usage. Perhaps there's another alternative that is more commonly used in the ecosystem.

Reply:
I think the pipeline syntax is good, but I'm a bit wary of adding this particular dependency, which appears to be quite a thin wrapper over lodash. This dependency has really small usage. Perhaps there's another alternative that is more commonly used in the ecosystem.

Reply:
@panuhorsmalahti hmm, interesting point. Here's some thoughts: 
1. We control this library ourselves, and for that, it is flexible and without magic. 

2. We created the library only recently, and therefore there is yet little adoption in the ecosystem.

3. We could have implemented this function in Lens as well, but it is pretty general to be local. Eg., also `injectable` uses it.

4. To look for an alternative full foreign library for this "thin wrapper" is a bit of an overkill, as the proposed pattern is universal, and does not require certain library to work properly with. Instead, any function with arity of one can participate in composition. Therefore a full alternate library would add extra weight with all the other bells and whistles.